torch_geometric/datasets/generators/graph_generator.py

```diff
@@ -0,0 +1,57 @@
+from typing import Callable, Optional
+
+import torch
+
+from torch_geometric.data import Data
+
+
+class GraphGenerator:
+    def __init__(self, motif: Optional[Callable], num_nodes: int = 300):
+        self.num_nodes = num_nodes
+        self.motif = motif
+        self.edge_index = None
+        self.edge_label = None
+        self.expl_mask = None
+        self.node_label = None
+        self.x = None
+
+    def generate_base_graph(self) -> Data:
+        raise NotImplementedError
+
+    # TODO: FeatureGenerator (feature distribution...)
+    def generate_feature(self, num_features: int = 10):
+        self.x = torch.ones((self.num_nodes, num_features), dtype=torch.float)
+
+    def attach_motif(self, num_motifs=80,
```

Comment:
we could have a test on attach_motif to e.g. make sure the correct number of motifs are attached (num nodes) etc.